packages/components/src/internal/components/buttons/ResponsiveMenuButtonGroup.tsx

```diff
@@ -9,45 +9,105 @@ interface ResponsiveMenuItem {
 }
 
 interface Props {
-    asSubMenu?: boolean;
     items: ResponsiveMenuItem[];
-    subMenuWidth?: number;
     user: User;
 }
 
+// The known size of the More dropdown. Ideally we'd calculate this, but we need to know it before we inject it, so
+// this was calculated by inspecting the DOM in FireFox.
+const MORE_SIZE = 70;
+
 export const ResponsiveMenuButtonGroup: FC<Props> = memo(props => {
-    const { items, user, subMenuWidth = 1600, asSubMenu = true } = props;
-    const [width, setWidth] = useState<number>(window.innerWidth);
-    useEffect(() => {
-        function handleResize() {
-            setWidth(window.innerWidth);
+    const { items, user } = props;
+    const [renderedItems, setRenderedItems] = useState<ReactElement[]>(items.map(item => item.button));
+    const [collapsedItems, setCollapsedItems] = useState<ReactElement[]>([]);
+    const [itemWidths, setItemWidths] = useState<number[]>([]);
+    const ref = useRef<HTMLElement>(undefined);
+    const computeButtonLayout = useCallback((): void => {
+        if (itemWidths.length === 0 || ref.current === undefined) {
+            // If we haven't determined itemWidths then we cannot properly determine how to render the buttons
+            return;
         }
-        window.addEventListener('resize', handleResize);
-        return () => window.removeEventListener('resize', handleResize);
-    });
+        const parent = ref.current.parentNode; // Should be responsive-btn-group, contains all the grid buttons
+        const grandParent = parent.parentNode as HTMLElement; // Should be button-bar__section, contains buttons + filter/search
+        const staticButtons = Array.from(parent.childNodes).reduce((reduction, node: HTMLElement) => {
+            if (!node.getAttribute('class').includes('responsive-menu-button-group')) {
+                reduction += node.getBoundingClientRect().width;
+            }
+            return reduction;
+        }, 0);
+        const filterAndSearch = grandParent.querySelector('.button-bar__filter-search').getBoundingClientRect().width;
```

Comment:
the `GridPanel.tsx` does have a `showFiltersButton` prop, so it seems like it is possible for this to come back without being able to find a filter-search component. null check needed?

Reply:
aww...nvm. I just saw how you wrapped this component below so maybe a null check is not needed.